src/middlewared/middlewared/plugins/pool_/dataset_details.py

```diff
@@ -156,6 +156,7 @@ class Config:
             'smb_shares': [],
             'iscsi_shares': [],
             'vms': [],
+            'virt_instances': [],
```

Comment:
I cannot comment on the above about service for some reason.
Please explain how we can not have the service running if we depend on the service to talk to incus abd know if there are containers?

Reply:
So the best way to go about it is that if the pool is set, we have incus running - and if it is unset, we don't have it running. This is sufficient to show that user explicitly enabled incus to run and should suffice for our use case as well

Reply:
The request was that people could interact with Incus entirely through the CLI id they want, this will make it more difficult, no?

Reply:
I don't think so ?
They explicitly set the pool via api/UI and that should be done regardless before we let the user use it so that we know incus state is properly initialized and contained in some pool.

Reply:
Why? They could do it all without UI support.

Reply:
We have never designed a feature/plugin to work with midcli. That's not a paradigm I'm interested in following any time soon. Especially since there has been discussion to remove that program entirely. We don't document it, we don't really test it, etc.

This feature should work the same as we do with apps. If user hasn't set a zpool to be used by incus, then we shouldn't start a bunch of unnecessary services.

Reply:
This is not midcli, its shell. It was a request from Kris. I will circle back with him about 5hat.